### DISCUSSION TITLE
Can hatch-mypyc also include dependencies?

### DISCUSSION THREAD

**Discussion by Tinche** [ofek/hatch-mypyc#22] [Q&A] [answered]
Hi!

I think this is a very interesting project. From having played around with mypyc, my understanding is that the greater the compilation unit, the greater the speedup (which makes sense, since early binding is more efficient and jumping in and out of Python is inefficient). So what I'm aiming for is, instead of expecting each library to build itself into mypyc wheels, it'd be better if libraries just provided annotated source code and the compilation should be done on the level of the app, including the app source code and all the dependencies.

So my question is, can hatch-mypyc include third party dependencies into the compilation unit?

I suppose at the start it'd need to be a whitelist of some sort since not all dependencies will be type-annotated. (I'm guessing there's no package-level metadata to automatically determine that?)

**Comment by ofek** [comment, marked as answer]
Not possible unless you vendor dependencies

**Reply by Tinche** [reply]
Ok, acknowledged. Would you consider adding this as a feature or do you think it's out of scope?

**Reply by ofek** [reply]
Out of scope since the better way is to vendor dependencies or if shipping Python run Mypyc on `site-packages`

**Reply by Tinche** [reply]
Alright, thanks.

**Reply by ofek** [reply]
Anytime!
